service/src/main/java/bio/terra/workspace/service/workspace/model/AwsCloudContext.java

```diff
@@ -97,6 +98,27 @@ public boolean equals(Object o) {
         && Objects.equal(commonFields, that.commonFields);
   }
 
+  /**
+   * Verifies that current cloud context is the same as the expected cloud context by compares only
+   * relevant fields
+   *
+   * @param expected {@link AwsCloudContext}
+   * @throws StaleConfigurationException StaleConfigurationException if they do not match
+   */
+  public void verifyCloudContext(AwsCloudContext expected) {
+    if ((this != expected)
```

Comment:
Its the objects are the same (i.e. variables point to same object in memory) it is clear that both variables refer to same object, and no further comparison is needed. 
in other words, checking the objects for '=' (memory address) is fail fast / short cut used for comparisons.

Reply:
Thought of that. In this particular case it is not needed. 
However I would recommend retaining it - 1) the function doesn't have visibility to who calls it, and where the params are from. 2) long shot - as a public function, it may be called from other locations which don't necessarily get it from the env again. 

If we want to remove the check, would suggest making it private and moving it to the same class as the caller. 
Note strongly opinionated either way. 

Generally i go for - 
- private functions: relaxed validation & checks since callers are relatively trustworthy in the same class. 
- protected functions: case by case determination
- public functions: more (reasonable) validation / checks